debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java

```diff
@@ -118,62 +103,106 @@ void close() {
         catch (Exception e) {
             LOGGER.warn("Exception while closing Jedis: {}", client, e);
         }
+        finally {
+            client = null;
+        }
+    }
+
+    /**
+    * Split collection to batches by batch size using a stream
+    */
+    private <T> Stream<List<T>> batches(List<T> source, int length) {
+        if (length <= 0) {
+            throw new IllegalArgumentException("length = " + length);
+        }
+
+        int size = source.size();
+        if (size <= 0) {
```

Comment:
```suggestion
        if (source.isEmpty()) {
```

Reply:
Thanks, I'll apply - in this case the size variable can be declared right after to avoid getting it without using it.